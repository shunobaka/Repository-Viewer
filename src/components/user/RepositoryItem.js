import React from 'react';
import { Accordion, Card } from 'react-bootstrap';
import PropTypes from 'prop-types';

const RepositoryItem = ({ repository, index }) => {
  return (
    <Card>
      <Accordion.Toggle as={Card.Header} eventKey={repository.id}>
        {`${index + 1}.`.padEnd(4)} {repository.name}
      </Accordion.Toggle>
      <Accordion.Collapse eventKey={repository.id}>
        <Card.Body>
          <a href={repository.url}>Go to repository</a>
          <br />
          <br />
          <h5>Description:</h5>
          {repository.description ? (
            <span>{repository.description}</span>
          ) : (
            <span className="text-warning">
              This repository does not have a description
            </span>
          )}
          <br />
          <br />
          <h5>Created on:</h5>
          <span>{new Date(repository.created_at).toLocaleDateString()}</span>
        </Card.Body>
      </Accordion.Collapse>
    </Card>
  );
};

RepositoryItem.propTypes = {
  repository: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    url: PropTypes.string,
    description: PropTypes.string,
    created_at: PropTypes.string,
  }),
  index: PropTypes.number.isRequired,
};

export default RepositoryItem;
