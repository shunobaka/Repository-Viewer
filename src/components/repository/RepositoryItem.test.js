/**
 * @fileoverview Defines snapshot unit tests for RepositoryItem component.
 */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RepositoryItem from './RepositoryItem';

describe('repositoryItem snapshots', () => {
  const mockRepo = {
    name: 'Test',
    id: 0,
  };

  it('repository renders correctly', () => {
    const index = 0;

    const tree = renderer
      .create(
        <BrowserRouter>
          <RepositoryItem repository={mockRepo} index={index} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
