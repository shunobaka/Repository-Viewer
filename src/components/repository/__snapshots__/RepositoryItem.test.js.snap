// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repositoryItem snapshots repository renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
    onClick={[Function]}
  >
    1.  
     
    Test
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    >
      <a>
        Go to repository
      </a>
      <br />
      <br />
      <h5>
        Description:
      </h5>
      <span
        className="text-warning"
      >
        This repository does not have a description
      </span>
      <br />
      <br />
      <h5>
        Created on:
      </h5>
      <span>
        Invalid Date
      </span>
    </div>
  </div>
</div>
`;
